application/src/main/java/org/togetherjava/tjbot/commands/moderation/meme/MemeListener.java

```diff
@@ -0,0 +1,65 @@
+package org.togetherjava.tjbot.commands.moderation.meme;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+
+import java.util.regex.Pattern;
+
+/**
+ * Listener that receives all sent messages from the Meme channel, checks if the message has media
+ * attched.
+ * <p>
+ * If there was no media attached, delete the messagen and send the User a DM, telling what they did
+ * wrong.
+ */
+public final class MemeListener extends MessageReceiverAdapter {
+
+    /**
+     * Creates a MemeListener to receive all message sent in Memes channel.
+     *
+     * @param config the config to use for this
+     */
+    public MemeListener(@NotNull Config config) {
+        super(Pattern.compile(config.getMemeChannelPattern()));
+    }
+
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+        boolean mediaAttachmentEmpty = isMediaAttachmentEmpty(event);
+        if (!mediaAttachmentEmpty) {
```

Comment:
Since its always used inverted, maybe invert the method to `hasMediaAttachments`.

That said, is the logic/naming correct? It appears that, right now, it reads as "if it has media, delete it"

Reply:
yeah you are right i changed the naming and logic